oak_functions/location_utils/src/lib.rs

```diff
@@ -14,239 +14,119 @@
+/// larger areas that fall outside of the zone.
 ///
-/// The cell with a western border at 0° longitude has a column number of 0. Column numbers increase
-/// eastward and cannot be smaller than 0.
-#[derive(Debug, Eq, PartialEq)]
-pub struct IndexKey {
-    pub row: i16,
-    pub col: i16,
-}
-
-impl IndexKey {
-    pub fn from_bytes(c: &[u8]) -> IndexKey {
-        let mut row_bytes = [0; 2];
-        row_bytes.copy_from_slice(&c[0..2]);
-        let mut col_bytes = [0; 2];
-        col_bytes.copy_from_slice(&c[2..4]);
-
-        IndexKey {
-            row: i16::from_be_bytes(row_bytes),
-            col: i16::from_be_bytes(col_bytes),
-        }
+/// See https://s2geometry.io/devguide/examples/coverings for more information on coverings.
+pub fn find_covering_cells(location: &LatLng, radius: &Angle, level: u8) -> Result<Vec<CellID>> {
+    if level > s2::cellid::MAX_LEVEL as u8 {
+        anyhow::bail!("level too high");
     }
-
-    /// Converts the `IndexKey` for the cell to bytes. This is used for looking up index entries for
-    /// a cell. Locations have 8 byte keys and cells have 4 byte keys, so there is no chance of a
-    /// collision.
-    pub fn to_bytes(&self) -> Vec<u8> {
-        [self.row.to_be_bytes(), self.col.to_be_bytes()].concat()
-    }
-}
-
-/// Represents an index value for a location-based lookup (e.g weather data).
-#[derive(Debug, Eq, PartialEq)]
-pub struct IndexValue {
-    /// The key for the location's related value (e.g. the current weather at the weather data
-    /// location).
-    pub value_key: [u8; 8],
-    /// The cartesian projection of its position relative to the midpoint of the cell.
-    pub position: Point,
+    let coverer = RegionCoverer {
+        min_level: level,
+        max_level: level,
+        level_mod: 0,
+        max_cells: 0,
+    };
+    let region = Cap::from_center_angle(&location.into(), radius);
+    Ok(coverer.covering(&region).0)
 }
 
-impl IndexValue {
-    pub fn from_bytes(c: &[u8]) -> IndexValue {
-        let mut value_key = [0; 8];
-        value_key.copy_from_slice(&c[0..8]);
-        IndexValue {
-            value_key,
-            position: Point::from_bytes(&c[8..16]),
-        }
+/// Finds the cell at the given `level` that covers the `location`.
+pub fn find_cell(location: &LatLng, level: u8) -> Result<CellID> {
+    let level = level as u64;
+    if level > s2::cellid::MAX_LEVEL {
+        anyhow::bail!("level too high");
     }
-
-    pub fn to_bytes(&self) -> Vec<u8> {
-        [self.value_key[..].to_vec(), self.position.to_bytes()].concat()
+    if level == s2::cellid::MAX_LEVEL {
+        return Ok(CellID::from(location));
     }
-}
 
-/// A relative location in cartesian coordinates.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct Point {
-    /// The relative x-coordinate in meters.
-    pub x: i32,
-    /// The relative y-coordinate in meters.
-    pub y: i32,
+    Ok(CellID::from(location).parent(level))
```

Comment:
What is `parent`? Can you add a  comment explaining why the parent is used in this case?